### DISCUSSION TITLE
Bluetooth Serial Connection between ConsolePi and  Iphone or iPad devices

### DISCUSSION THREAD

**Discussion by wissamoh** [Pack3tL0ss/ConsolePi#126] [Q&A]
Hi, Has anybody succeeded to get and iPhone or iPad connect to console terminal through bluetooth? if yes how is it possible?
I Connected my iPad to the consolePI through bluetooth...but what next? how can I get a terminal connection to the consolePI?

**Comment by Pack3tL0ss** [comment]
I'll try to provide more detail once I get to my ipad, but you would:

1. pair / connect to ConsolePi / bluetooth
2. Use a bluetooth serial / terminal app from the appstore to connect it should open up a CLI and provide a simple menu

I can't recall exactly which app I used will try to provide more detail when I have my ipad handy.

**Reply by wissamoh** [reply]
Hi,
Thanks for your reply, Actually I already tried these 2 steps...I paired the PI with my iPad and tried many serial and terminal apps but all of these where bound to specific product vendors...Thus I had no luck to make it work. 
I apreciate if you can share or recall the method you used to get it work.

Thanks in advance.
Wissam

**Reply by staffordrau** [reply]
In case anyone is looking to do this with something running MacOS, I found it very easy to get my 2019 macbook working with it. On the Pi zero w I'm using, I Googled the basic cli bluetooth commands:

consolepi01# bluetoothctl
agent on
agent-default
discoverable on

It then showed up in my macbook's bluetooth settings, so I selected it to pair it. On the pi it asked me if I wanted to pair with device X with code XXXXXX, so I answered yes.

On the macbook, I'm using zoc8 as my serial terminal app. In the zoc8 host directory, I created a new connection. Under "Connection type" I selected Serial/Direct. I then hit the "Configure..." button next to the "Connection type". For the "Com-Port" setting I hit the "Scan..." button, and it found /dev/cu.consolepi01, which I selected. Changed the speed to the standard 9600, and that was it. I was able to open that connection and connect to the consolepi and use the two usb/serial ports on it remotely.

Surprisingly easy and straightforward, probably more so than under Windows.

**Reply by rudivandrunen** [reply]
Hi all, 
Just pretty new to ConsolePi, but I cannot get my bluetooth connection working to my mac (seems that to a windows machine I have the very same issue). Details:
- MacOS Seqoia 15.2 Macbook Pro M1
- ConsolePi Version: 2024-3.6 Installer Version 84
- Rpi ZeroW Linux consolepi 6.6.67+ #1833 Fri Dec 20 14:07:35 GMT 2024 armv6l GNU/Linux

OK what happens is following.... I see consolepi in my mac bt pairing menu, but after pair attempt, it throws an error (no details). When using the terminal prg "serial" (decisivetactics.com) I see the consolepi in the bt menu, and can connect, but it just does not give a prompt.(neither on 9600 or 115200bd, no characters whatsoever) 
On windows I just get an error when trying to pair :-(  seems to have to do with SPP and pairing. Also my iPhone (ios18) fails in pairing.
ok, as bt is not my core business, I tried to experiment a bit with the rfcomm service and bt settings, (is this rfcomm detecting the pair and is the agetty started ? but still no success. 
So, also no bt logging on the consolepi. 

any tips ?? Also, how do I get logging enabled to further troubleshoot the bt .....
Also: tips on how

**Reply by Pack3tL0ss** [reply]
@rudivandrunen I'm not a mac guy, but it should be similar to what I do to pair it to another Linux system.

It's essentially the process highlighted in the previous post by staffordau
On the ConsolePi

- Enter command `groups` to check if the user is a member of the `bluetooth` group
  - If `bluetooth` is not listed add the user to the bluetooth group `sudo usermod -a -G bluetooth <USERNAME>`
- Go into bluetooth utility `bluetoothctl`
- Enter the following commands
  - discoverable on
  - pairable on
  _Stay in `bluetoothctl`_
 
Then on the mac search for devices, you should see the ConsolePi.  Attempt to pair with it, it should see a confirmation on the ConsolePi asking you to approve the pairing request.  Follow the prompt to accept it, and you should be paired/able to connect.

Hopefully that helps
